app/vmalert/datasource/datasource.go

```diff
@@ -24,6 +24,30 @@ type Metric struct {
 	Value     float64
 }
 
+// MetricSample is used for templating annotations,
+// Labels as map simplifies templates evaluation.
+type MetricSample struct {
```

Comment:
Should we change `type Metric` instead?

Reply:
I'm not happy with public type breaking changes.

 Another option, introduce private type for notifier package and make conversion there?